### PR TITLE
[admin] added full screen mode to advanced json editor

### PR DESCRIPTION
**Pull request by gastonche**
PR pertaining to issue #42.

### CODE REVIEW

**Comment by gastonche** [comment]
@nemesisdesign  I submitted the PR already and would like you to review it so I can correct any issues you may find soon.

**Comment by nemesisdesign** [comment]
Testing now.

**Comment by nemesisdesign** [comment]
I would say we are halfway through it. In order for this to be really full screen, a few slightly more complex things should be implemented:

- the height of the advanced editor should adapt to the height of the window, eg: `div.height($(window).height())`
- the height of the advanced editor should change if the window is resized (bind to resize event of window object)
- the main scrollbars of the body element must be disabled (set `overflow: hidden`)
- when back to normal mode, the height of the advanced editor must be cleaned up, the `overflow: hidden` directive also must be cleaned up and the resize event must be unloaded

There are also a few graphical improvements I want to suggest:

- in full screen mode, the help text "configuration in NetJSON DeviceConfiguration format" and its container (with white background) can be hidden so that the editor can become prevalent

**Comment by gastonche** [comment]
Okay, I'll update and submit a PR by night. @nemesisdesign

**Comment by nemesisdesign** [comment]
@gastonche you can update this PR by amending your commit (`git --amend`) and using `git push -f` to your own development branch

**Comment by gastonche** [comment]
Okay, thanks for the tip @nemesisdesign

**Comment by gastonche** [comment]
@nemesisdesign I have done effected all of the changes you specified. But given that the editor in full screen mode has to hide all other elements, is it sufficient that i have enabled the use of the ESC key as a way to exit full screen mode or should I also add some more controls the the top of the screen hanging over the editor to enable the user exit full screen mode by clicking?

**Comment by nemesisdesign** [comment]
@gastonche I don't know because I haven't seen the result. Don't be afraid to push your work even if unfinished. Push it, I'll test it and come back to you with some useful suggestions.

File: django_netjsonconfig/static/django-netjsonconfig/js/widget.js [hunk 1]
```diff
@@ -1,4 +1,37 @@
...
+                $('#advanced_editor').height($(window).height()).width(window.outerWidth);
+            });
+            inFullScreenMode = true;
+            // Listen for escape key to leave full screen mode
+            $(document).keyup(function(e) {
+                 if (e.keyCode == 27) { 
+                    toggleFullScreen();
+                }
+            });
+        }else{
```
Comment (nemesisdesign) [inline django_netjsonconfig/static/django-netjsonconfig/js/widget.js]: for consistency with the rest of the code, better to have

```
}
else {
```

File: django_netjsonconfig/static/django-netjsonconfig/js/widget.js [hunk 2]
```diff
@@ -1,4 +1,37 @@
...
+            $('#advanced_editor').removeClass('full-screen').height(oldHeight).width(oldWidth);
+            $('body').removeClass('editor-full');
+            // unbind all events listened to while going to full screen mode
+            $(window).unbind('resize');
+            $(document).unbind('keyup');
+            inFullScreenMode = false;
+            document.getElementById('advanced_editor').scrollIntoView(true);
+        }
+
```
Comment (nemesisdesign) [inline django_netjsonconfig/static/django-netjsonconfig/js/widget.js]: delete blank line

**Comment by nemesisdesign** [comment]
Try to squash the 4 commits into one by using `git rebase -i HEAD~4`

**Comment by gastonche** [comment]
okay, I'll do that, I just wanted you to test that before i finally did

**Comment by nemesisdesign** [comment]
I will test now ;-)

**Comment by nemesisdesign** [comment]
@gastonche always ping me when you do changes, we are getting a lot of work and sometimes I may miss some details. Don't be afraid to ping me once in a while.

**Comment by gastonche** [comment]
Okay, I have been offline for some days now, but i'll fix that this
evening. Sorry about the delay

**Comment by gastonche** [comment]
@nemesisdesign I think we are Good to Go, I have done the required fixes and updated the PR

**Comment by gastonche** [comment]
@nemesisdesign I think we are good to go. I have done all the requested changes for this PR and updated the PR as requested.

**Comment by nemesisdesign** [comment]
@gastonche excellent. Very good job. I am merging but I will do a few style changes myself and I would like you to take a look at them so that in the future you will be able to do this kind of things without my intervention.

**Comment by nemesisdesign** [comment]
@gastonche I added some improvements 0dc39c4 and created issue #45 after testing the patch, there are few issues that will have to be resolved before it can be released

**Comment by gastonche** [comment]
Okay, let me refresh my fork and check out the changes you made

**Pull request merged**